Okta.Core/OktaHttpClient.cs

```diff
@@ -175,6 +176,29 @@ public override Task<HttpResponseMessage> ExecuteAsync(HttpRequestType requestTy
                     {
                         httpClient.DefaultRequestHeaders.Remove("Authorization");
                     }
+                    if (bAddXForwardedHeader)
+                    {
+                        string ipAddress = null;
+                        if (!string.IsNullOrEmpty(content))
+                        {
+                            content = content.Trim();
+                            try
+                            {
+                                var parsedContent = JObject.Parse(content);
+                                ipAddress = (string)parsedContent["context"]["ipAddress"];
+                            }
+                            catch (Exception e)
+                            {
+                                /* no ipAddress given */
+                                throw new OktaException("The AuthnRequest object is malformed." + e.Message);

```

Comment:
Is there ever a situation where the OktaHttpClient will POST some content that isn't JSON? (Maybe `application/x-www-form-urlencoded`? ping @jmelberg-okta )
If so, it will always throw an error here now.